app/controllers/app/freestyle_controller.rb

```diff
@@ -4,9 +4,13 @@ class App::FreestyleController < App::BaseController
   FILTER_OPTIONS = Exercise.distinct.pluck(:body_part).freeze
 
   # renders all the exercises with that particular body part
+  # also renders all the exercises that match a particular search param
   def index
    apply_filter(params[:filter_by]) 
-   apply_search(params[:search])  
+
+   if params[:search] != nil 
+    apply_search(params[:search])  
+   end
```

Comment:
That's good! This solve the problem.

But imagine that we need to reuse the method `apply_search` in another context.
We probably would need to repeat this guard clause against `nil` again, right?

Also, that search feature is something that relates to the database and it is being handled in the controller, that seems odd to me.

What do you think about the interface below?

```ruby
def index
  Exercise.search(params[:search])
end
```

Using this simple interface, the controller doesn't know any implementation detail about how the records are filtered or ordered. The controller just passes the parameters under the `search` key to the `Exercise.search` method. No private methods in the controller to handle that, no constants. The controller is a thinner layer now.

If you agree with this approach, do you think you can make the necessary changes to make it work like that?

Reply:
I don't think we will need to run the nil check again in another context, why i am running it here is because both apply_filter and apply_search are in the same action and return the same variable **@freestyle_exercises** , so anytime we want to run the apply_filter only, the apply_search also runs but with a nil params thereby returning all the exercises in the db.

In another context, i have passed the **required** html attribute to the form tag to check if it is nil before it even sends the post request.

I will move the search logic to the exercise model in subsequent pull requests.

Thank you.